src/main/java/io/sqooba/traildb/TrailDBIterator.java

```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
             public boolean hasNext() {
                 TrailDBEvent next = TrailDBNative.INSTANCE.cursorNext(TrailDBIterator.this.cursor);
                 if (next == null) {
+                    // Auto-close when at the end because it is not intuitive to have to close an
+                    // iterator when at its end in Java.
+                    TrailDBIterator.this.close();
```

Comment:
Does this fix the leak you had (i.e. issue #3 ) ?
BTW, I'm not really understanding your comment... what do you mean by "at the end" ? at the end of the iteration process ?